### DISCUSSION TITLE
Do song ids retain across server restarts

### DISCUSSION THREAD

**Discussion by itslychee** [MusicPlayerDaemon/MPD#1899] [Q&A] [answered]
I believe the title should explain it all

**Comment by MaxKellermann** [comment]
What is a "song id"?

**Reply by itslychee** [reply]
As it's described here https://mpd.readthedocs.io/en/latest/protocol.html#the-queue

> Song ids on the other hand are stable: an id is assigned to a song when it is added, and will stay the same, no matter how much it is moved around. Adding the same song twice will assign different ids to them, and a deleted-and-readded song will have a new id. This way, a client can always be sure the correct song is being used.

**Comment by naglis** [comment, marked as answer]
After a quick look at https://github.com/MusicPlayerDaemon/MPD/blob/feac1a3f56591ccfb00870aba66c9ac3abac4773/src/queue/QueueSave.cxx#L51 and https://github.com/MusicPlayerDaemon/MPD/blob/feac1a3f56591ccfb00870aba66c9ac3abac4773/src/SongSave.cxx#L52 and a simple test with `addid`/`deleteid`/`playlistid` commands across MPD restarts it seems that song ids are not written to the state file and thus are not preserved across MPD restarts.

**Reply by itslychee** [reply]
Thanks for the detailed response! Thankfully I wrote my program to work with any eventual MPD restart so this shouldn't be a problem.
